client/web-sveltekit/src/routes/[...repo=reporev]/(validrev)/(code)/FileTree.svelte

```diff
@@ -125,7 +123,12 @@
                     tabindex={-1}
                     data-go-up={isRoot ? true : undefined}
                 >
-                    <Icon svgPath={getIconPath(entry, expanded)} inline />
+                    {#if entry.isDirectory}
+                        <Icon svgPath={getIconPath(entry, expanded)} inline />
+                    {:else}
+                        {@const fileInfo = getFileInfo(entry.name)}
+                        <Icon svgPath={fileInfo.icon.svgPath} inline --color={fileInfo.icon.color} />
```

Comment:
Tiny nit: I think you could do this if you want to make it more compact (I like shorthand props 🤷🏻 )

```suggestion
                        {@const {icon: {svgPath, color}} = getFileInfo(entry.name)}
                        <Icon {svgPath} inline --color={color} />
```